File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/schedule/ScheduleRule.java
package com.example.chat.domain.schedule;

import java.time.Instant;

import com.example.chat.domain.message.Message;

/**
 * 테스트에서 사용하는 최소한의 ScheduleRule 구현
 */
public record ScheduleRule(
        ScheduleType type,
        Message message,
        Instant scheduledAt,
        CronExpression cronExpression) {

    // JavaBean 호환 getters
    public ScheduleType getType() { return type(); }
    public Message getMessage() { return message(); }
    public Instant getScheduledAt() { return scheduledAt(); }
    public CronExpression getCronExpression() { return cronExpression(); }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/schedule/ScheduleRule.kt
package com.example.chat.domain.schedule

import com.example.chat.domain.message.Message
import java.time.Instant

/**
 * 테스트에서 사용하는 최소한의 ScheduleRule 구현
 */
class ScheduleRule(
    private val type: ScheduleType,
    private val message: Message,
    private val scheduledAt: Instant?,
    private val cronExpression: CronExpression?
) {
    fun getType(): ScheduleType = type
    fun getMessage(): Message = message
    fun getScheduledAt(): Instant? = scheduledAt
    fun getCronExpression(): CronExpression? = cronExpression
}
